platforms/software/software-diagnostics/src/main/java/org/gradle/api/plugins/ProjectReportsPlugin.java

```diff
@@ -24,11 +24,14 @@
 import org.gradle.internal.deprecation.DeprecationLogger;
 import org.gradle.util.internal.WrapUtil;
 
+import javax.annotation.Nonnull;
+
 /**
  * <p>A {@link Plugin} which adds some project visualization report tasks to a project.</p>
  *
  * @see <a href="https://docs.gradle.org/current/userguide/project_report_plugin.html">Project Report plugin reference</a>
  */
+@Nonnull
```

Comment:
It seems this class is a mixture of base and software report tasks. 
Should we split this class up into base/software?

Reply:
I agree, tasks, projects, properties should move to base.  I made a coarse first cut and didn't want to divide individual classes wherever possible.  I'll make a note to add this to a followup PR.